### PR TITLE
fix: buffered future blocks group removal [WPB-28040]

### PR DESCRIPTION
**Pull request by coriolinus**
We should only restore buffered messages if we are still a member of the group at that point.

### CODE REVIEW

File: crypto/src/mls/conversation/mutable/decrypt/buffer_messages.rs [hunk 1]
```diff
@@ -334,6 +334,87 @@ mod tests {
...
+    ///
+    /// This is ordinary out-of-order delivery, the very thing the buffering machinery exists for:
+    /// the DS hands us an epoch `n + 1` application message before it hands us the epoch `n + 1`
+    /// commit which removed us. Once that commit arrives, the conversation must be wiped, and the
+    /// buffered message must go with it. If instead the buffered message is replayed against the
+    /// new epoch, decryption fails identically on every retry, in every subsequent transaction, so
+    /// the client can never process its own removal and the conversation is stuck forever.
+    #[apply(all_cred_cipher)]
+    async fn buffered_message_must_not_block_own_removal(case: TestContext) {
+        Box::pin(async move {
```
Comment (SimonThormeyer) [inline crypto/src/mls/conversation/mutable/decrypt/buffer_messages.rs]: I know this is done in many other places, but we probably don't need to `Box::pin` the entire test to avoid a stack overflow here?

Comment (coriolinus) [reply crypto/src/mls/conversation/mutable/decrypt/buffer_messages.rs]: ~~Doesn't overflow on my machine; or CI (so far); I'd only do it if we have evidence it's necessary.~~ Oops, misunderstood your comment.

Comment (coriolinus) [reply crypto/src/mls/conversation/mutable/decrypt/buffer_messages.rs]: Dropped the pinning en-passant while rebasing onto main.

File: crypto/src/mls/conversation/mutable/decrypt/mod.rs [hunk 2]
```diff
@@ -358,7 +358,7 @@ impl ConversationMut {
 
                 // can't use `.then` because async
                 let mut buffered_messages = None;
-                if recursion_policy == RecursionPolicy::AsNecessary {
+                if is_active && recursion_policy == RecursionPolicy::AsNecessary {
```
Comment (SimonThormeyer) [inline crypto/src/mls/conversation/mutable/decrypt/mod.rs:361]: Wanna copy and paste the commit messasge body here, for context why this is needed?

**Review by SimonThormeyer — APPROVED** [review]
lgtm!
